### DISCUSSION TITLE
Replacing min() and max() for DCTL

### DISCUSSION THREAD

**Discussion by nmbr73** [nmbr73/Shadertoys#27] [Q&A] [answered]
We replace `float max(float,float)` with DCTL's `float _fmaxf(float, float)`.

@J-i-P-i why did you suggest to replace it with `float max_f(float,float)`- did I get something wrong?!?

But my actual question is: Why is it, that `float min(float,float)` does not need to be substituted? I have overseen an occurence of `min()`in FractalLand.fuse, but it runs on all plattforms?!?

**Comment by J-i-P-i** [comment, marked as answer]
The origin is in the Readme from Blackmagic in (Partion): \ ProgramData \ Blackmagic Design \ DaVinci Resolve \ Support \ Developer \ DaVinciCTL \ README.txt

All mathematical functions and their "appearance" are given there.
I have to admit, I have done this conversion very "blindly" so far, but have also seen that e.g. the function `clamp ()` works without modification.

I have now specifically examined the `max ()` in Cuda:

 - with float you can use max (float)

- if more dimensions are added, the `_fmaxf ()` is required for Cuda, which was not so clear to me until now. It still has to be tested whether OpenCL is also supported for _fmaxf float2 to float4.


Now, I tested max() in OpenCL and there are no Convertions necessary, ->

### So the change in `_fmaxf()` is only necessary for Cuda, but the explicit distinctions `max_f ()` can be omitted
